modules/common/systemd/base.nix

```diff
@@ -0,0 +1,295 @@
+      "rpcbind.target"
+      "systemd-update-done.service"
+      "system-update.target"
+      "system-update-pre.target"
+      "system-update-cleanup.service"
+    ]
+    ++ (lib.optionals (!cfg.withMachines) [
+      "container-getty.service"
+      "container-getty@.service"
+      "container@.service"
+      "systemd-nspawn@.service"
+    ])
+    ++ (lib.optionals ((!cfg.withDebug) && (!cfg.withSerial)) [
+      "getty.service"
+      "getty@.service"
+      "getty.target"
+      "getty-pre.target"
+      "serial-getty.service"
+      "serial-getty@.service"
+      "serial-getty.target"
+      "serial-getty@.target"
+    ])
+    ++ (lib.optionals ((!cfg.withDebug) && (!cfg.withJournal)) [
+      "systemd-journald-audit.socket"
+      "systemd-journal-catalog-update.service"
+      "systemd-journal-flush.service"
+      "systemd-journald.service"
+      "systemd-journald@.service"
+      "systemd-journal-gatewayd.socket"
+      "systemd-journald-audit.socket"
+      "systemd-journald-dev-log.socket"
+      "systemd-journald-varlink@.socket"
+      "systemd-journald.socket"
+      "systemd-journald@.socket"
+      "systemd-update-utmp.service"
+    ])
+    ++ (lib.optionals (!cfg.withAudit) [
+      "audit.service"
+      "auditd.service"
+      "systemd-journald-audit.socket"
+    ])
+    ++ (lib.optionals ((!cfg.withDebug) && (!cfg.withMachines)) [
+      "systemd-coredump.socket"
+    ])
+    ++ (lib.optionals (!cfg.withLogind) [
+      "systemd-logind.service"
+      "dbus-org.freedesktop.login1.service"
+    ])
+    ++ (lib.optionals (!cfg.withNss) [
+      "nscd.service"
+      "nss.service"
+      "nss.target"
+      "nss-lookup.target"
+      "nss-user-lookup.target"
+      "nss-lookup.target.requires"
+      "nss-user-lookup.target.requires"
+    ])
+    ++ (lib.optionals (!cfg.withTimesyncd) [
+      "systemd-timesyncd.service"
+    ])
+    ++ (lib.optionals (!cfg.withResolved) [
+      "systemd-resolved.service"
+    ])
+    ++ (lib.optionals (!cfg.withNetworkd) [
+      "network.target"
+      "network-pre.target"
+      "network-online.target"
+      "network-interfaces.target"
+      "network-setup.service"
+      "network-local-commands.service"
+      "systemd-timesyncd.service"
+      "systemd-networkd-wait-online.service"
+      "systemd-networkd.service"
+      "systemd-networkd.socket"
+    ])
+    ++ (lib.optionals (!cfg.withDebug) [
+      ## Units kept with debug
+      "kbrequest.target"
+      "rescue.service"
+      "rescue.target"
+      "emergency.service"
+      "emergency.target"
+      "systemd-vconsole-setup.service"
+      "reload-systemd-vconsole-setup.service"
+      "console-getty.service"
+      "sys-kernel-debug.mount"
+      "sys-fs-fuse-connections.mount"
+      "systemd-pstore.service"
+      "mount-pstore.service"
+      "systemd-ask-password-console.path"
+      "systemd-ask-password-console.service"
+      "systemd-ask-password-wall.path"
+      "systemd-ask-password-wall.service"
+    ]);
```

Comment:
Regarding lines https://github.com/tiiuae/ghaf/pull/506/files#diff-8799260fcdef43f44d9f5a097ac22bbc9122cada55afae08610edf81ccf480fbR53-R163
I feel the `optionals` lists of units to be removed could be refactored into files that are then imported for unit removal from each VM. This is kind of opinionated comment as I find (multiple) exclusive conditions a tad hard to read because one would have to go to the VM declarations to check which config options apply to which VM. If the systemd unit exclusions were imported and declared in the VMs, it could be clearer. We did conceptually something similar with hardened kernel configuration fragments (separate files) based on config option declarations.

So, I guess I'm writing this down as something to consider in further iterations.